framework/utils/randomUtils.py

```diff
@@ -0,0 +1,244 @@
+    if rawInt < low or rawInt > high:
+      caller.raiseAMessage("Random int out of range")
+      rawInt = max(low,min(rawInt,high))
+    return rawInt
+
+def randomPermutation(l,caller):
+  """
+    Function to get a random permutation
+    @ In, l, list, list to be permuted
+    @ In, caller, instance, the caller
+    @ Out, newList, list, randomly permuted list
+  """
+  if stochasticEnv == 'numpy':
+    return npStochEnv.permutation(l)
+  else:
+    newList = []
+    oldList = l[:]
+    while len(oldList) > 0:
+      newList.append(oldList.pop(randomIntegers(0,len(oldList)-1,caller)))
+    return newList
+
+def randPointsOnHypersphere(dim,samples=1,r=1,keepMatrix=False):
+  """
+    obtains random points on the surface of a hypersphere of dimension "n" with radius "r".
+    see http://www.sciencedirect.com/science/article/pii/S0047259X10001211
+    "On decompositional algorithms for uniform sampling from n-spheres and n-balls", Harman and Lacko, 2010, J. Multivariate Analysis
+    @ In, dim, int, the dimensionality of the hypersphere
+    @ In, samples, int, optional, the number of samples desired
+    @ In, r, float, optional, the radius of the hypersphere
+    @ In, keepMatrix, bool, optional, if True then will always return np.array(np.array(float))
+    @ Out, pts, np.array(np.array(float)), random points on the surface of the hypersphere [sample#][pt]
+  """
+  ## first fill random samples
+  pts = randomNormal(dim,samples=samples,keepMatrix=True)
+  ## extend radius, place inside sphere through normalization
+  rnorm = float(r)/np.linalg.norm(pts,axis=1)
+  pts *= rnorm[:,np.newaxis]
+  #TODO if all values in any given sample are 0,
+  #       this produces an unphysical result, so we should resample;
+  #       however, this probability is miniscule and the speed benefits of skipping checking loop seems worth it.
+  if keepMatrix:
+    return pts
+  else:
+    return _reduceRedundantListing(pts,dim,samples)
+  return pts
+
+def randPointsInHypersphere(dim,samples=1,r=1,keepMatrix=False):
+  """
+    obtains a random point internal to a hypersphere of dimension "n" with radius "r"
+    see http://www.sciencedirect.com/science/article/pii/S0047259X10001211
+    "On decompositional algorithms for uniform sampling from n-spheres and n-balls", Harman and Lacko, 2010, J. Multivariate Analysis
+    @ In, dim, int, the dimensionality of the hypersphere
+    @ In, r, float, the radius of the hypersphere
+    @ In, keepMatrix, bool, optional, if True then will always return np.array(np.array(float))
+    @ Out, pt, np.array(float), a random point on the surface of the hypersphere
+  """
+  #sample on surface of n+2-sphere and discard the last two dimensions
+  pts = randPointsOnHypersphere(dim+2,samples=samples,r=r,keepMatrix=True)[:,:-2]
+  if keepMatrix:
+    return pts
+  else:
+    return _reduceRedundantListing(pts,dim,samples)
+  return pts
+
+### internal utilities ###
+
+def _reduceRedundantListing(data,dim,samples):
```

Comment:
It appears you can get similar behavior with reshape, except that you can't reshape an array to get a float.  I don't see a meaningful way to rewrite this method using reshape.